indico/modules/categories/operations.py

```diff
@@ -113,6 +117,10 @@ def _log_category_update(category, changes, extra_log_fields):
             'type': 'text',
             'convert': lambda changes: [_format_wallet_credentials(x) for x in changes],
         },
+        'apple_pass_mode': 'Apple Pass mode',
+        'apple_pass_certificate': {'title': 'Apple Pass certificate', 'type': 'string'},
+        'apple_pass_key': {'title': 'Apple Pass private key', 'type': 'string'},
+        'apple_pass_password': {'title': 'Apple Pass certificate password', 'type': 'string'},
```

Comment:
I don't think we should even be storing it at all: Why not decrypt the key when saving it, and store it in standard unencrypted PEM format? If someone dumps he DB they'd have both key and encrypted file anyway.

Reply:
I do not understand what do you want me to do here.

Reply:
Right now you store the encrypted cert and the corresponding passphrase in the settings. I think it'd be better to use the passphrase *while saving* to decrypt it, and then only store the decrypted certificate (in standard ASCII PEM format, instead of binary)

Reply:
@OmeGak @ThiefMaster 
I can think about 2-3 different way of achieving the password field not to be saved, one of those is about:

- remove the `apple_wallet_password` from `APPLE_WALLET_JSON_FIELDS` 
- add `apple_wallet_password` to update_category skip parameters (`update_category(self.category, form.data, skip={'meeting_theme', 'lecture_theme', 'apple_wallet_password'})`)
- remove `DataRequired()` from `apple_wallet_password` in the form

If you do not like this approach, please let me know.

Reply:
I think the cleanest option - but probably out of scope here - would be to have a custom field there where you can drop the file and enter the key. Existing data would just be indicated as "key uploaded on [date]" or something similar w/ the option to replace or remove it.

Unfortunately pretty much all the options are slightly ugly simple because this is always a workflow that WTForms is not ideal for (it's best for cases where you enter data, save it as entered, and show it as saved whenrendering the form again). But as soon as files are involved... eh, not so great.

So I think your suggestion of making the password field (and file upload field as well, I assume) optional sounds good. I would then add some custom validation that fails when:

- a password is provided but no key uploaded [makes no sense to do that]
- a key is uploaded which is encrypted, but the password is missing / invalid

Regarding your original question I think doing it on the WTForm level is the cleanest option, ie you validate/decrypt the key there and the data that "leaves" the form is already the plain version.

Note: You'll nonetheless need to come up with something to indicate that there is currently a key saved... I'd be OK with a slightly ugly workaround of using the description of the key file upload field for this (you can modify that via `self.fieldname.description` in the constructor after the super call).

Reply:
- (and file upload field as well, I assume) 
Why? it can remain mandatory, I'll just replace the content with the decrypted value (PS: both `apple_wallet_key` and `apple_wallet_certificate` are `TextAreaField`)

Reply:
How would you save other data in that form if there's a mandatory file upload field? You're not going to upload a file unless you want to change the key...

Reply:
Oh, I wasn't aware it's a text field. I somehow thought the key you get from apple is in some binary format, and not ascii.

Reply:
I'm fine with storing both the key and passphrase in that case. But I still think the passphrase should not be required (as long as validation fails if they key is encrypted)... Unless you think that they key is always encrypted anyway (I don't think "user locally decrypts it manually" is a relevant case here).

Reply:
OK, so I'll keep everything as is (`apple_wallet_password` in the `APPLE_WALLET_JSON_FIELDS`) , but I remove DataRequired() from apple_wallet_password in the form.
I'll keep the private key encrypted as it is submitted.
Is that ok?

Reply:
yep